File: php-laravel/lib/app/Models/RequestRoutesArrivalSearch.php
<?php
/**
 * RequestRoutesArrivalSearch
 */
namespace app\Models;

/**
 * RequestRoutesArrivalSearch
 */
class RequestRoutesArrivalSearch {

    /** @var string $id */
    private $id;

    /** @var string[] $departureLocationIds */
    private $departureLocationIds;

    /** @var string $arrivalLocationId */
    private $arrivalLocationId;

    /** @var \app\Models\RequestTransportation $transportation */
    private $transportation;

    /** @var \DateTime $arrivalTime */
    private $arrivalTime;

    /** @var \app\Models\RequestRoutesProperty[] $properties */
    private $properties;

    /** @var \app\Models\RequestRangeFull $range */
    private $range;

}

File: fsharp-functions/OpenAPI/src/model/RequestRoutesArrivalSearch.fs
namespace OpenAPI.Model

open System
open System.Collections.Generic
open Newtonsoft.Json
open OpenAPI.Model.RequestRangeFull
open OpenAPI.Model.RequestRoutesProperty
open OpenAPI.Model.RequestTransportation

module RequestRoutesArrivalSearch = 

  //#region RequestRoutesArrivalSearch

  [<CLIMutable>]
  type RequestRoutesArrivalSearch = {
    [<JsonProperty(PropertyName = "id")>]
    Id : string;
    [<JsonProperty(PropertyName = "departure_location_ids")>]
    DepartureLocationIds : string[];
    [<JsonProperty(PropertyName = "arrival_location_id")>]
    ArrivalLocationId : string;
    [<JsonProperty(PropertyName = "transportation")>]
    Transportation : RequestTransportation;
    [<JsonProperty(PropertyName = "arrival_time")>]
    ArrivalTime : DateTime;
    [<JsonProperty(PropertyName = "properties")>]
    Properties : RequestRoutesProperty[];
    [<JsonProperty(PropertyName = "range")>]
    Range : RequestRangeFull;
  }
  
  //#endregion
  